tsinfer/inference.py

```diff
@@ -696,21 +649,46 @@ def insert_sites(self, tables):
             tree = next(trees)
             for variant in self.sample_data.variants():
                 site = variant.site
+                predefined_anc_state = site.ancestral_state
                 while tree.interval[1] <= site.position:
                     tree = next(trees)
                 assert tree.interval[0] <= site.position < tree.interval[1]
                 tables.sites.add_row(
                     position=site.position,
-                    ancestral_state=site.ancestral_state,
+                    ancestral_state=predefined_anc_state,
                     metadata=self.encode_metadata(site.metadata))
                 if site.inference == 1:
                     tables.mutations.add_row(
                         site=site.id, node=node[inferred_site],
                         derived_state=variant.alleles[derived_state[inferred_site]])
                     inferred_site += 1
                 else:
-                    assert ts.num_edges > 0
-                    self.locate_mutations_on_tree(tree, variant, tables.mutations)
+                    if np.all(variant.genotypes == tskit.MISSING_DATA):
```

Comment:
There's bunch of stuff about missing data that I don't understand here. If you'd like this merged quickly then you'll need to remove the stuff relating to missing data --- since there's no missing data support in tskit right now, then by necessity this is currently untested. Otherwise, if you don't want to do this, you may as well roll it in with the missing data update. But, I'll need to spend a day working through this to figure out all the consequences.

Reply:
Well, this part is simply checking for the extreme case of completely missing data in a non-inference site, which would otherwise cause map_mutations to fail. I can remove it for this PR, and just assume that we'll never hit that issue if we aren't allowing missing data input in the first place.

Reply:
As I said, up to you to do play it, if you want to remove the missing data references from here to get this merged quickly, then great.